src/components/elements/PathHandler.tsx

```diff
@@ -0,0 +1,31 @@
+import { ReactNode, useEffect, useRef } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+
+const PathHandler = ({ children }: { children: ReactNode }) => {
+  const location = useLocation();
+  const navigate = useNavigate();
+  const attempts = useRef(0);
+
+  const { pathname, hash, search } = location;
+  const match = pathname.match(/^(.*)(%20)+$/);
+
+  useEffect(() => {
+    if (attempts.current >= 5) {
+      console.error(`Unable to correct route '${pathname}'`);
```

Comment:
I'm not quite following the "attempts", how exactly does this code get hit? Is it trying to make it so the redirect only happens on the first 5 renders? (I couldn't repro this error log)

I think we only really need to do this redirect when someone is loading the page for the first time. So perhaps another approach is would be to only run the `useEffect` once, what do you think of that? seems easier to reason about to me:
```
  useEffect(() => {
    if (match) {
      const [, corrected] = match;
      attempts.current += 1;
      navigate({ pathname: corrected, hash, search }, { replace: true });
    }

    // ignore pathname dependency, we only want to run this once
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return children;
```

Reply:
Technically the attempts should never increment since the page should be redirected on the first try, I just put it there to prevent infinite navigation attempts if, for some reason, the page wasn't able to redirect to a different URL. But yeah, I agree that just running it once is probably cleaner and should solve the problem. I'll make that change